### PR TITLE
[editor] Take into account created features while saving.

### PR DESCRIPTION
**Pull request by mgsergio**
@VladiMihaylenko

### CODE REVIEW

File: map/framework.cpp [hunk 1]
```diff
@@ -2669,6 +2669,10 @@ osm::Editor::SaveResult Framework::SaveEditedMapObject(osm::EditableMapObject em
   // Notify if a poi address and it's hosting building address differ.
   do
   {
+    // A feature was created. Nothing to note.
+    if (emo.GetID().IsValid())
```
Comment (deathbaba) [inline map/framework.cpp]: Эта проверка возвращает true если фича внутри mwm. Видимо, пропущено отрицание !

Comment (mgsergio) [inline map/framework.cpp]: Спасибо)

Comment (mgsergio) [inline map/framework.cpp]: Done.

**Comment by VladiMihaylenko** [comment]
LGTM

File: map/framework.cpp [hunk 2]
```diff
@@ -2737,7 +2767,7 @@ osm::Editor::SaveResult Framework::SaveEditedMapObject(osm::EditableMapObject em
                       editedFeature.GetHouseNumber() != emo.GetHouseNumber()) ||
                      (editor.GetEditedFeatureStreet(emo.GetID(), editedFeatureStreet) &&
                       !editedFeatureStreet.empty() &&
-                      editedFeatureStreet != emo.GetStreet().m_defaultName);
+                      editedFeatureStreet != emo.GetStreet().m_defaultName) && !isCreatedFeature;
```
Comment (deathbaba) [inline map/framework.cpp]: Эту проверку лучше вставить в начало, а не в конец.

File: map/framework.cpp [hunk 3]
```diff
@@ -2712,11 +2735,18 @@ osm::Editor::SaveResult Framework::SaveEditedMapObject(osm::EditableMapObject em
       shouldNotify = true;
     }
 
+    // TODO(mgsergio): Consider this:
+    // A user changed a feature with no original addres and addres differ from the one
```
Comment (deathbaba) [inline map/framework.cpp]: User has changed feature with no original address, and entered address is different from the one

File: map/framework.cpp [hunk 4]
```diff
@@ -2712,11 +2735,18 @@ osm::Editor::SaveResult Framework::SaveEditedMapObject(osm::EditableMapObject em
...
+    // on a hosting building. He saved it. This was uploaded to OSM. Then he set
```
Comment (deathbaba) [inline map/framework.cpp]: It was uploaded into OSM.

File: map/framework.cpp [hunk 5]
```diff
@@ -2712,11 +2735,18 @@ osm::Editor::SaveResult Framework::SaveEditedMapObject(osm::EditableMapObject em
...
+    // back an address from a hosting building. As a result this address woun't
```
Comment (deathbaba) [inline map/framework.cpp]: will not be merged into OSM, because we clear street and house number in the following code.

**Comment by deathbaba** [comment]
LGTM после минорных изменений.

File: map/framework.cpp (hunk 2 above)
Comment (mgsergio) [inline map/framework.cpp]: Done.

File: map/framework.cpp (hunk 3 above)
Comment (mgsergio) [inline map/framework.cpp]: Done.

File: map/framework.cpp (hunk 4 above)
Comment (mgsergio) [inline map/framework.cpp]: Done.

**Pull request merged**